src/main/java/com/cloudbees/plugins/credentials/CredentialsDescriptor.java

```diff
@@ -192,4 +257,339 @@ public String getIconClassName() {
     public boolean isApplicable(CredentialsProvider provider) {
         return true;
     }
+
+    /**
+     * In some cases the nearest {@link AncestorInPath} {@link ModelObject} is one of the Credentials plugins wrapper
+     * classes.
+     * This helper method unwraps those to return the correct context.
+     *
+     * @param context the context (wrapped or unwrapped).
+     * @return the unwrapped context.
+     * @since 2.1.5
+     */
+    @NonNull
+    public static ModelObject unwrapContext(@NonNull ModelObject context) {
+        if (context instanceof CredentialsSelectHelper.WrappedCredentialsStore) {
+            return ((CredentialsSelectHelper.WrappedCredentialsStore) context).getStore().getContext();
+        }
+        if (context instanceof CredentialsStoreAction.CredentialsWrapper) {
+            return ((CredentialsStoreAction.CredentialsWrapper) context).getStore().getContext();
+        }
+        if (context instanceof CredentialsStoreAction.DomainWrapper) {
+            return ((CredentialsStoreAction.DomainWrapper) context).getStore().getContext();
+        }
+        return context;
+    }
+
+    /**
+     * Looks up the context given the provider and token.
+     * @param provider the provider.
+     * @param token the token.
+     * @return the context.
+     *
+     * @since 2.1.5
+     */
+    public static ModelObject lookupContext(String provider, String token) {
```

Comment:
Undocumented null (no Javadoc/Annotation). Even if it's a common behavior for this method, better to require checks